theano/tensor/raw_random.py

```diff
@@ -133,15 +133,28 @@ def __init__(self, fn, outtype, inplace=False, ndim_added=0):
         self.__setstate__([fn, outtype, inplace, ndim_added])
 
     def __getstate__(self):
-        return self.state
+        d = dict(self.__dict__)
+        del d['exec_fn']
+        if 'destroy_map' in d:
+            del d['destroy_map']
+        return d
+
+    def __setstate__(self, dct):
+        if isinstance(dct, dict):
+            state = [dct['fn'],
+                     dct['outtype'],
+                     dct['inplace'],
+                     dct['ndim_added']]
+        else:
+            state = dct
```

Comment:
This is still not ok since it will still delete any attributes of the superclass.